backend/lm_backend/api/routes/jobs.py

```diff
@@ -96,55 +139,69 @@ async def delete_job(
 
 
 @router.delete(
-    "/slurm_job_id/{slurm_job_id}/cluster/{cluster_id}",
+    "/slurm_job_id/{slurm_job_id}",
     status_code=status.HTTP_200_OK,
 )
 async def delete_job_by_slurm_id(
     slurm_job_id: str,
-    cluster_id: int,
     secure_session: SecureSession = Depends(secure_session(Permissions.JOB_EDIT)),
 ):
     """
     Delete a job from the database and associated bookings.
 
-    Uses the slurm_job_id and the cluster_id to filter the job.
+    Uses the slurm_job_id and the cluster client_id to filter the job.
 
-    Since the slurm_job_id can be the same across clusters, we need the cluster_id to validate.
+    Since the slurm_job_id can be the same across clusters, we need the cluster client_id to validate.
     """
-    jobs: List[Job] = await crud_job.read_all(
-        db_session=secure_session.session, search=slurm_job_id, force_refresh=True
+    client_id = secure_session.identity_payload.client_id
+
+    if not client_id:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=("Couldn't find a valid client_id in the access token."),
+        )
+
+    jobs: List[Job] = await crud_job.filter(
```

Comment:
could you add a filter term for `cluster_client_id` to this filter method?

Reply:
I updated the CRUD filter method to accept a list of expressions to filter.
Now it's possible filter using both parameters at the same time.